pinot-common/src/main/java/com/linkedin/pinot/common/http/MultiGetRequest.java

```diff
@@ -95,24 +89,22 @@ public MultiGetRequest(@Nonnull Executor executor,
    * @return instance of CompletionService. Completion service will provide
    *   results as they arrive. The order is NOT same as the order of URLs
    */
-  public CompletionService<GetMethod> execute(@Nonnull List<String> urls, final int timeoutMs) {
-    Preconditions.checkNotNull(urls);
-    Preconditions.checkArgument(timeoutMs > 0, "Timeout value for multi-get must be greater than 0");
+  public CompletionService<GetMethod> execute(List<String> urls, int timeoutMs) {
+    HttpClientParams clientParams = new HttpClientParams();
+    clientParams.setConnectionManagerTimeout(timeoutMs);
+    HttpClient client = new HttpClient(clientParams, _connectionManager);
 
-    CompletionService<GetMethod> completionService = new ExecutorCompletionService<>(executor);
-    for (final String url : urls) {
-      completionService.submit(new Callable<GetMethod>() {
-        @Override
-        public GetMethod call()
-            throws Exception {
-          HttpClient client = new HttpClient(connectionManager);
+    CompletionService<GetMethod> completionService = new ExecutorCompletionService<>(_executor);
+    for (String url : urls) {
+      completionService.submit(() -> {
+        try {
           GetMethod getMethod = new GetMethod(url);
           getMethod.getParams().setSoTimeout(timeoutMs);
-          // if all connections in the connection manager are busy this will wait to retrieve a connection
-          // set time to wait to retrieve a connection from connection manager
-          client.getParams().setConnectionManagerTimeout(timeoutMs);
           client.executeMethod(getMethod);
           return getMethod;
+        } catch (Exception e) {
+          LOGGER.warn("Caught {} while executing GET on URL: {}", e.getClass().getName(), url);
```

Comment:
Might be better to log e.getMessage().

Reply:
Changed to e.toString() so that class and message are both logged